src/promptflow-core/promptflow/_utils/flow_utils.py

```diff
@@ -107,6 +110,10 @@ def resolve_flow_path(
                 f"Please specify a file or remove the extra YAML file.",
                 privacy_info=[str(flow_path)],
             )
+        elif allow_prompty_dir and check_flow_exist:
+            candidates = list(flow_folder.glob(f"*{PROMPTY_EXTENSION}"))
+            if len(candidates) == 1:
+                flow_file = candidates[0].name
```

Comment:
There can be more than one file with .prompty extension. The returned file path could not be expected.

Reply:
offline synced. This is related to the basic interface of current serve app constructor and it's dangerous to change this near release cut.